clients/admin-ui/src/features/datastore-connections/ConnectionTypeLogo.tsx

```diff
@@ -11,14 +12,18 @@ import {
 import { DatastoreConnection, isDatastoreConnection } from "./types";
 
 type ConnectionTypeLogoProps = {
-  data: string | DatastoreConnection;
+  data: string | DatastoreConnection | ConnectionSystemTypeMap;
 };
 
 const ConnectionTypeLogo: React.FC<ConnectionTypeLogoProps & ImageProps> = ({
   data,
   ...props
 }) => {
   const getImageSrc = (): string => {
+    if (isConnectionSystemTypeMap(data) && data.encoded_icon) {
+      return `data:image/svg+xml;base64,${data.encoded_icon}`;
```

Comment:
Expanding `data` to include `ConnectionSystemTypeMap` so we can pull the `encoded_icon` for SaaS connectors